### PR TITLE
Fix NRE initializing views

### LINKED ISSUE #93: NRE when starting the GUI
I just tried to compile the GUI from master and ran into a NRE when starting the gui.

image

Stacktrace
```csharp
System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=TestCentric.Gui.Runner
  StackTrace:
   at TestCentric.Gui.Controls.TestSuiteTreeView.set_CheckBoxes(Boolean value) in C:\src\NUnit\test-centric-gui\src\TestCentric\testcentric.gui\Controls\TestSuiteTreeView.cs:line 172
   at TestCentric.Gui.Controls.TestTree.set_ShowCheckBoxes(Boolean value) in C:\src\NUnit\test-centric-gui\src\TestCentric\testcentric.gui\Controls\TestTree.cs:line 123
   at TestCentric.Gui.Controls.TestTree.InitializeView(ITestModel model, TestCentricPresenter presenter) in C:\src\NUnit\test-centric-gui\src\TestCentric\testcentric.gui\Controls\TestTree.cs:line 718
   at TestCentric.Gui.TestCentricPresenter.InitializeControls(Control owner) in C:\src\NUnit\test-centric-gui\src\TestCentric\testcentric.gui\TestCentricPresenter.cs:line 478
   at TestCentric.Gui.TestCentricPresenter.InitializeControls(Control owner) in C:\src\NUnit\test-centric-gui\src\TestCentric\testcentric.gui\TestCentricPresenter.cs:line 480
   at TestCentric.Gui.TestCentricPresenter.OnStartup() in C:\src\NUnit\test-centric-gui\src\TestCentric\testcentric.gui\TestCentricPresenter.cs:line 97
   at TestCentric.Gui.TestCentricMainForm.NUnitForm_Load(Object sender, EventArgs e) in C:\src\NUnit\test-centric-gui\src\TestCentric\testcentric.gui\TestCentricMainForm.cs:line 1100
   at System.Windows.Forms.Form.OnLoad(EventArgs e)
...
```
The problem is that `Model` is null in `                    Model.Services.UserSettings.Gui.TestTree.ShowCheckBoxes = base.CheckBoxes = value;` as `InitializeView` has not been called yet. I think that this was introduced by #89

### PR DESCRIPTION
**Pull request by CharliePoole**
Fixes #93 

This is the short-term fix described in the issue.

### CODE REVIEW

**Comment by CharliePoole** [comment]
@mikkelbu @immeraufdemhund I'd appreciate a review from one of you. This is no more than a bandaid, I'm afraid. @mikkelbu For some reason I can't use the dropdown to request your review.

**Comment by CharliePoole** [comment]
The test would be based on the repo, of course, and I couldn't repo it. What we are trying to test is essentially a negative... that the ShowCheckBoxes setter is not called by TestTree before all initialization is complete. Since I removed the setter and the call, I think it's more a question of whether that broke anything else.

I'm inclined to move into restructuring this to use pure MVP next, adding tests of the individual presenters.

File: src/TestCentric/testcentric.gui/Controls/TestSuiteTreeView.cs [hunk 1]
```diff
@@ -169,7 +169,9 @@ public new bool CheckBoxes
                         ? new VisualState(this)
                         : null;
 
-                    Model.Services.UserSettings.Gui.TestTree.ShowCheckBoxes = base.CheckBoxes = value;
+                    // TODO: Get rid of updatimg model from the view. Till we do, this needs to be guarded.
```
Comment (mikkelbu) [inline src/TestCentric/testcentric.gui/Controls/TestSuiteTreeView.cs]: Major nitpick: updatimg => updating

**Comment by CharliePoole** [comment]
Merging and moving on. See issue #95.

**Pull request merged**